### DISCUSSION TITLE
Failed to apply extern force to body

### DISCUSSION THREAD

**Discussion by zhang-qiang-github** [InteractiveComputerGraphics/PositionBasedDynamics#158] [Q&A]
`setAcceleration(particleIndex, accel).` may apply extern force to a body.

But, in the `TimeStepController::step`, all force is clear:

```

void TimeStepController::step(SimulationModel &model)
{
	START_TIMING("simulation step");
	TimeManager *tm = TimeManager::getCurrent ();
	const Real hOld = tm->getTimeStepSize();
 
	//////////////////////////////////////////////////////////////////////////
	// rigid body model
	//////////////////////////////////////////////////////////////////////////
	clearAccelerations(model);
        ...
}
```
So, how can I apply an extern force?

**Comment by janbender** [comment]
I think an easy solution would be to use your own TimeStepController. Just write a new class MyTimeStepController which is derived from the class TimeStepController and implement a new step method. In this step method you can add any forces after the clearAcceleration command. 

Hope that helps!
